hi-ml-histopathology/src/histopathology/utils/layer_utils.py

```diff
@@ -23,7 +23,16 @@ def setup_feature_extractor(pretrained_model: nn.Module,
         feature_extractor = pretrained_model
     except AttributeError:
         # Otherwise fallback to sequence of child modules:
-        layers = list(pretrained_model.children())[:-1]
+        layers = list(pretrained_model.children())
+        if len(layers) <= 1:
+            # If `layers` contains only one element (the entire model)
+            # Fetch the model and find its children (individual layers)
+            # Remove the last classification layer
+            layers = list(list(pretrained_model.children())[0].children())[:-1]
+        else:
+            # If `layers` contains multiple elements, these are the individual model layers
+            # Remove the last classification layer
+            layers = list(pretrained_model.children())[:-1]
```

Comment:
good catch - can we extend the test to capture this case? this will ensure we don't encounter the same bug in the future (and it will avoid we decrease test coverage with this PR)

Reply:
@vale-salvatelli  To capture this case, a test needs to be added for the SSL encoder in `test_encoders.py`. Not sure why this was initially not added, is it because it requires a checkpoint path to download the weights? Which checkpoint path should be given here, can a test SSL run be used?

Reply:
could you check with Melissa? she has maybe a test run that could be used. Otherwise, instead of downloading, we could patch the SSL encoder with an encoder initialized in the test - we don't care in this case if the weights are random but just that we're selecting the right layer

Reply:
Thanks. I have added a test for the SSL encoder.